### ISSUE TITLE
Multithreading: Run initial page load on the thread pool

### ISSUE THREAD

**Issue by ratanasov** [holoviz/panel#5861]
#### Is your feature request related to a problem? Please describe.
Pages loading data on initial page load (without deferring and async) don't seem to take any advantage of enabling multithreading (`pn.config.nthrads`).

#### Describe the solution you'd like
The initial page load code is run on the thread pool, so multiple users don't block each other when loading the page.

#### Describe alternatives you've considered
This feature request is part of an evaluation of Panel concurrency options, including deferring, multiprocessing and async.

#### Additional context
@philippjfr said in a call this sounds like a reasonable feature request.

#### Code Sample
```py
import panel as pn
import time
import threading

def load(event=None):
    tid = threading.get_ident()

    print(f"[{time.time():.3f}] loading on {tid} ...")
    time.sleep(5)
    print(f"[{time.time():.3f}] done loading on {tid}")

    return "# app"

def app():
    btn = pn.widgets.Button(name="reload", on_click=load)
    return pn.Column(load, btn)

if __name__ == "__main__":
    pn.config.nthreads = 4

    pn.serve(app)
```

Output during (almost simultaneous) initial page load in two browser tabs shows that the page loads are run sequentially on the same thread:
```
[1699875470.234] loading on 7951815424 ...
[1699875475.239] done loading on 7951815424
[1699875475.311] loading on 7951815424 ...
[1699875480.316] done loading on 7951815424
```

Output during (almost simultaneous) reload button clicks in two browser tabs shows that the callbacks are run concurrently on different threads:
```
[1699875561.193] loading on 6325039104 ...
[1699875562.219] loading on 6341865472 ...
[1699875566.198] done loading on 6325039104
[1699875567.224] done loading on 6341865472
```

**Comment by philippjfr** [comment]
Was looking into this over the weekend. It should not be impossible but my initial attempts on this weren't successful.

**Comment by MarcSkovMadsen** [comment]
+1: The first user in my ai app will trigger a load of an embedding index. Takes 30secs to load. During this it blocks the main thread.

**Comment by philippjfr** [comment]
>The first user in my ai app will trigger a load of an embedding index. Takes 30secs to load. During this it blocks the main thread

In that case you should `--warm` your server.

**Comment by MarcSkovMadsen** [comment]
Yes

**Comment by philippjfr** [comment]
I'm still not following. We are talking about server startup.

**Comment by MarcSkovMadsen** [comment]
Yes. Users dont like their app to be unavailable. We might be redeploying multiple times a day during working hours. I need to minimize the risk of unavailability.

**Comment by philippjfr** [comment]
Got you, I'm used to rolling restarts where Kubernetes waits for the new container to respond to a liveness check before it shuts down the old deployment and the ingress controller swaps them out transparently.

**Comment by MarcSkovMadsen** [comment]
And probably I also need to work on getting that in order 😄
